src/mongo/tree/MongoAccountTreeItem.ts

```diff
@@ -43,11 +44,24 @@ export class MongoAccountTreeItem implements IAzureParentTreeItem {
     public async loadMoreChildren(_node: IAzureNode, _clearCache: boolean): Promise<IAzureTreeItem[]> {
         let db: Db | undefined;
         try {
+            let databases: IDatabaseInfo[];
+
             db = await MongoClient.connect(this.connectionString);
-            const result: { databases: IDatabaseInfo[] } = await db.admin().listDatabases();
-            return result.databases
+            let databaseInConnectionString = getDatabaseNameFromConnectionString(this.connectionString);
+            if (databaseInConnectionString && !this.isEmulator) { // emulator violates the connection string format
+                // If the database is in the connection string, that's all we connect to (we might not even have permissions to list databases)
```

Comment:
I'm fine with this as long as the connection strings from Azure do not list a database by default. Is that the case? I just want to make sure Azure Cosmos DB account still list all databases

Reply:
That's the case for both connected and attached scenarios.